hs_core/hydroshare/hs_bagit.py

```diff
@@ -55,40 +54,42 @@ def delete_bag(resource):
 
 def create_bag_files(resource, fed_zone_home_path=''):
     """
-    create and update all files needed by bagit operation that is conducted on iRODS server; no bagit operation
-    is performed, only files that will be included in the bag are created or updated.
+    create and update files needed by bagit operation that is conducted on iRODS server; no bagit
+    operation is performed, only files that will be included in the bag are created or updated.
 
     Parameters:
-    :param resource: A resource whose files will be created or updated to be included in the resource bag.
-    :param fed_zone_home_path: Optional, A passed-in non-empty value indicates the resource needs to be created
-    in a federated zone rather than in the default hydroshare zone.
-    :return: istorage, an IrodsStorage object,that will be used by subsequent operation to create a bag on demand as needed.
+    :param resource: A resource whose files will be created or updated to be included in the
+    resource bag.
+    :param fed_zone_home_path: Optional, A passed-in non-empty value that indicates that the
+    resource needs to be created in a federated zone rather than in the default hydroshare zone.
+    :return: istorage, an IrodsStorage object that will be used by subsequent operation to
+    create a bag on demand as needed.
     """
     from hs_core.hydroshare.utils import current_site_url, get_file_mime_type
-    DATE_FORMAT = "YYYY-MM-DDThh:mm:ssTZD"
 
     if fed_zone_home_path:
-        istorage=IrodsStorage('federated')
+        istorage = IrodsStorage('federated')
     else:
-        istorage=IrodsStorage()
-
-    dest_prefix = getattr(settings, 'BAGIT_TEMP_LOCATION', '/tmp/hydroshare/')
-    bagit_path = os.path.join(dest_prefix, resource.short_id, arrow.get(resource.updated).format("YYYY.MM.DD.HH.mm.ss"))
-
-    for d in (dest_prefix, bagit_path):
-        try:
-            os.makedirs(d)
-        except OSError as ex:
-            if ex.errno == errno.EEXIST:
-                shutil.rmtree(d)
-                os.makedirs(d)
-            else:
-                raise Exception(ex.message)
+        istorage = IrodsStorage()
+
+    dest_prefix = os.path.join(getattr(settings, 'IRODS_ROOT', '/tmp'), 'hydroshare')
+    bagit_path = os.path.join(dest_prefix, resource.short_id,
+                              arrow.get(resource.updated).format("YYYY.MM.DD.HH.mm.ss"))
```

Comment:
@pkdash It already uses uuid on the path, the date value is a sub-directory under uuid directory just to be more "unique", and it is a temp folder which will be deleted after the bag is transferred to iRODS side, so it should be good enough.